verbose: true
debug: true
auto_pause_rosbag: false

submap_creation_interval: 10
loop_closure_topic_queue_length: 1000
future_loop_closure_queue_length: 10
submap_topic_queue_length: 10
publisher_queue_length: 10

input_odom_frame: "odom"
input_base_link_frame: "imu"
output_odom_frame: "odom"
output_base_link_frame: "imu"
output_sensor_frame: "sensor_voxgraph"

# T_base_sensor:
# - [-0.9908275, -0.0196803, 0.1336922, -0.024]
# - [ 0.0202159, -0.9997921, 0.0026504, -0.011]
# - [ 0.1336122,  0.0053288, 0.9910194, -0.131]
# - [ 0.0,        0.0,       0.0,        1.0]

measurements:
  submap_registration:
    enabled: false
    sampling_ratio: 0.05
    registration_method: "explicit_to_implicit"
    information_matrix:
      x_x:     1.0
      y_y:     1.0
      z_z:     1.0
      yaw_yaw: 1.0
  odometry:
    enabled: true
    information_matrix:
      x_x:     1000.0
      y_y:     1000.0
      z_z:     2500.0
      yaw_yaw: 2500.0
  height:
    enabled: false
    information_zz: 2500.0
  loop_closure:
    enabled: true
    information_matrix:
      x_x:     100.0
      y_y:     100.0
      z_z:     250.0
      yaw_yaw: 250.0


mesh_min_weight: 2.0
submap_mesh_color_mode: "lambert_color"
combined_mesh_color_mode: "lambert_color"

tsdf_voxel_size: 0.10
truncation_distance: 0.30
max_weight: 1000
pointcloud_queue_size: 10
max_ray_length_m: 10 # 14 or 16m would be better, but for that we need better odometry. Otherwise the surface get too damaged.
min_ray_length_m: 0.2
max_consecutive_ray_collisions: 10
use_const_weight: true
use_weight_dropoff: false
use_sparsity_compensation_factor: true
sparsity_compensation_factor: 20.0

esdf_max_distance: 4
esdf_min_distance: 0.1

publish_maps_every_n_sec: 1.0
loop_closure_topic: "/loop_closure_in"

vis_combined_o3d_mesh: false
k_traj: 0.0
k_overlap: 0.0
